### PR TITLE
Controller vault route

### PR DESCRIPTION
**Pull request by guyblink**
Integrations fetch credentials by controller endpoint. No more vault direct access.
BE changes should be merged prior to this. Plugins should be updated to latest blink-sdk version.

### CODE REVIEW

File: plugin/connections/connection_instance.go [hunk 1]
```diff
@@ -26,61 +21,36 @@ type ConnectionInstance struct {
...
+	marshalledData, err := json.Marshal(connectionData)
 
-	client, err := c.acquireVaultAPIConnection()
 	if err != nil {
 		return nil, err
 	}
 
-	vaultSecretPath := fmt.Sprintf(vaultSecretPathFormat, c.Name, c.Id)
-	secrets, err := client.Logical().Read(vaultSecretPath)
+	secretRequest, err := http.NewRequest(http.MethodGet, c.VaultUrl, bytes.NewBuffer(marshalledData))
```
Comment (havivrosh) [inline plugin/connections/connection_instance.go:38]: put all the http part behind one method: getToken or getSecret or something similar.

Comment (havivrosh) [inline plugin/connections/connection_instance.go:38]: make it post or if you want to make it get, do it with params. something like this: /vault?connectionTypeKey=xxx& connectionNameKey=yyy...

I prefer post here.

File: plugin/connections/connection_instance.go [hunk 2]
```diff
@@ -26,61 +21,36 @@ type ConnectionInstance struct {
...
-		err = fmt.Errorf("completed the request successfully but no secrets returned with path %s", vaultSecretPath)
-		log.Error(err)
-		return nil, err
-	}
+	secretResponse, err := client.Do(secretRequest)
 
-	if secrets.Data == nil {
-		err = errors.New("invalid secret structure, data map missing")
-		log.Error(err)
+	body, err := ioutil.ReadAll(secretResponse.Body)
```
Comment (havivrosh) [inline plugin/connections/connection_instance.go:46]: you need to close the body

**Review by havivrosh — CHANGES_REQUESTED** [review]

File: plugin/connections/connection_instance.go (hunk 1 above)
Comment (guyblink) [reply plugin/connections/connection_instance.go:38]: Yes. Fixed to POST

**Pull request merged**